File: ch9/randomizer.php
<?php 
// how many rows should be in each of the test tables?
$table_target_size = 1000;
     
// connect to db, set up query, run the query
$dbc = mysqli_connect('localhost','username','password','stackoverflow')
       or die('Error connecting to database!' . mysqli_error());
$dbc->set_charset("utf8");
       
$tables = array("badges",
    "comments",
    "posts",
    "post_history",	
    "post_links",
    "tags",
    "users",
    "votes");

foreach ($tables as $table)
{   
	echo "\n=== Now working on $table ===\n";
    $select_table_info = "SELECT count(Id) as c, min(Id) as mn, max(Id) as mx FROM $table";
    $table_info = mysqli_query($dbc, $select_table_info);
    $table_stuff = mysqli_fetch_object($table_info);
    $table_count = $table_stuff->c;
    $table_min = $table_stuff->mn;
    $table_max = $table_stuff->mx;
    
    // set up loop to grab a random row and insert into new table
    $i=0;
    while($i < $table_target_size)
    {
        $r = rand($table_min, $table_max);
        echo "\nIteration $i: $r";
        $insert_rowx = "INSERT IGNORE INTO test_$table (SELECT * FROM $table WHERE Id = $r)";
        $current_row = mysqli_query($dbc, $insert_rowx);
       
        $select_current_count = "SELECT count(*) as rc FROM test_$table";
        $current_count= mysqli_query($dbc, $select_current_count);
        $row_count = mysqli_fetch_object($current_count)->rc;
        $i = $row_count;
    }
}
?>

File: ch9/randomizer.py
# -*- coding: utf-8 -*-
import sys
import random
import pymysql

password = sys.argv[1]
tableTargetSize = 1000;

# Open local database connection
db = pymysql.connect(host='localhost',
                     db='stackoverflow',
                     user='',
                     passwd=password,
                     port=3306,
                     charset='utf8mb4',
                     autocommit=True)
cursor = db.cursor()
cursor1 = db.cursor()

# this is a hack to get around Python not allowing me to execute dynamic tables
# reference: http://stackoverflow.com/questions/26497846/python-mysql-parameter-queries-for-dynamic-table-names

tableList = ['SELECT count(Id), min(Id), max(Id) FROM badges',
    'SELECT count(Id), min(Id), max(Id) FROM comments',
    'SELECT count(Id), min(Id), max(Id) FROM posts',
    'SELECT count(Id), min(Id), max(Id) FROM post_history',
    'SELECT count(Id), min(Id), max(Id) FROM post_links',
    'SELECT count(Id), min(Id), max(Id) FROM tags',
    'SELECT count(Id), min(Id), max(Id) FROM users',
    'SELECT count(Id), min(Id), max(Id) FROM votes']
    
testQueryList = ['INSERT IGNORE INTO test_badges SELECT * FROM badges WHERE Id = %s',
    'INSERT IGNORE INTO test_comments SELECT * FROM comments WHERE Id = %s',
    'INSERT IGNORE INTO test_posts SELECT * FROM posts WHERE Id = %s',
    'INSERT IGNORE INTO test_post_history SELECT * FROM post_history WHERE Id = %s',
    'INSERT IGNORE INTO test_post_links SELECT * FROM post_links WHERE Id = %s',
    'INSERT IGNORE INTO test_tags SELECT * FROM tags WHERE Id = %s',
    'INSERT IGNORE INTO test_users SELECT * FROM users WHERE Id = %s',
    'INSERT IGNORE INTO test_votes SELECT * FROM votes WHERE Id = %s']
    
testCountList = ['SELECT count(*) FROM test_badges',
    'SELECT count(*) FROM test_comments',
    'SELECT count(*) FROM test_posts',
    'SELECT count(*) FROM test_post_history',
    'SELECT count(*) FROM test_post_links',
    'SELECT count(*) FROM test_tags',
    'SELECT count(*) FROM test_users',
    'SELECT count(*) FROM test_votes']
i=0
for tableQuery in tableList:
    print("\n=== Now working on", tableQuery, "(test table ",i,")")
    
    try:
        cursor.execute(tableQuery)
        result = cursor.fetchone()
    except:
        print(cursor._last_executed)
        raise
    tableCount = result[0]
    tableMin = result[1]
    tableMax = result[2]
    
    # set up loop to grab a random row and insert into new table
    j=0;
    while j < tableTargetSize:

        r = random.randrange(tableMin,tableMax)
        print("Iteration",j,":",r);
        
        cursor1.execute(testQueryList[i],(r,));
        cursor1.execute(testCountList[i]);
        currentCount = cursor1.fetchone()[0]
        j = currentCount
        
    i = i+1
db.close()
